invenio_banners/services/service.py

```diff
@@ -47,3 +74,41 @@ def create(self, identity, data, raise_errors=True):
         return self.result_item(
             self, identity, banner, links_tpl=self.links_item_tpl, errors=errors
         )
+
+    def delete(self, identity, id):
+        """Delete a banner from database."""
+        self.require_permission(identity, "delete")
+
+        banner = self._read(id)
+
+        self.record_cls.delete(banner)
+
+        return self.result_item(self, identity, banner, links_tpl=self.links_item_tpl)
+
+    def update(self, identity, id, data):
+        """Update a banner."""
+        self.require_permission(identity, "update")
+
+        banner = self._read(id)
+
+        self.record_cls.update(data, id)
+
+        return self.result_item(
+            self,
+            identity,
+            banner,
+            links_tpl=self.links_item_tpl,
+        )
+
+    def disable_expired(self, identity):
+        """Disable expired banners."""
+        self.require_permission(identity, "disable")
+        self.record_cls.disable_expired()
+
+    def _read(self, id):
```

Comment:
Eventually, you can have the `Record.get` to execute `query.one()`, which will raise if not existing, and remove this method.